runtime/dart_isolate.cc

```diff
@@ -195,10 +198,12 @@ std::weak_ptr<DartIsolate> DartIsolate::CreateRunningRootIsolate(
     root_isolate_create_callback();
   }
 
-  if (!isolate->RunFromLibrary(dart_entrypoint_library,       //
-                               dart_entrypoint,               //
-                               settings.dart_entrypoint_args  //
-                               )) {
+  const std::vector<std::string>& args = !dart_entrypoint_args.empty()
+                                             ? dart_entrypoint_args
+                                             : settings.dart_entrypoint_args;
```

Comment:
This is unfortunate.  How is a developer supposed to know their `settings.dart_entrypoint_args` is being overwritten.  Wouldn't it be better to merge the arguments?

Reply:
In the current implementation details, `RunConfiguration::dart_entrypoint_args` will only be used on the Mobile platform , while `Settings::dart_entrypoint_args` will only be used on the Desktop platform and the Fuchsia platform, so there will not be a situation where both of them are set.

In addition, we are planning to remove `Settings::dart_entrypoint_args`. Here is the issue I filed: https://github.com/flutter/flutter/issues/93459

So I guess there should be fine if we not merge them. Please let me know if this makes sense for you.

Reply:
SGTM, I'd throw an assert in there if they can't both be set.

Reply:
Done